lib/createCookieEncryptor.js

```diff
@@ -0,0 +1,73 @@
+import base64url from 'base64url'
+
+const KEY_ALGORITHM = 'AES-GCM'
+
+const toHexString = bytes =>
+  bytes.reduce((str, byte) => str + byte.toString(16).padStart(2, '0'), '')
+
+const fromHexString = hexString =>
+  new Uint8Array(hexString.match(/.{1,2}/g).map(byte => parseInt(byte, 16)))
+
+export const createCookieEncryptor = (cookieName, encryptionKey) => {
+  const getKey = async () => {
+    try {
+      return crypto.subtle.importKey(
+        'raw',
+        base64url.toBuffer(encryptionKey),
+        { name: KEY_ALGORITHM },
+        false,
+        ['encrypt', 'decrypt']
+      )
+    } catch (e) {
+      console.error({ getKeyError: e })
+    }
+  }
+
+  const encrypt = async str => {
+    try {
+      const key = await getKey(KEY_ALGORITHM)
+      const iv = crypto.getRandomValues(new Uint8Array(16))
+      const encoder = new TextEncoder()
+      const data = encoder.encode(str)
+      const encrypted = await crypto.subtle.encrypt({ name: KEY_ALGORITHM, iv }, key, data)
+      return `${toHexString(iv)}${base64url(encrypted)}`
+    } catch (e) {
+      console.error({ encryptError: e })
+    }
+  }
+
+  const decrypt = async str => {
+    try {
+      const iv = fromHexString(str.slice(0, 32))
+      const token = str.slice(32)
+      const key = await getKey(KEY_ALGORITHM)
+      const decrypted = await crypto.subtle.decrypt({
+        name: KEY_ALGORITHM, iv
+      }, key, base64url.toBuffer(token))
+      const decoder = new TextDecoder()
+      return decoder.decode(decrypted)
+    } catch (e) {
+      console.error({ decryptError: e })
+    }
+  }
+
+  const decryptionHandler = async (request, event, config) => {
+    const { headers } = request
+    try {
+      const token = headers.get(cookieName)
+      if (!token) {
+        return new Response('no cookie present', { status: 500 })
+      }
+      const decrypted = await decrypt(token)
+      headers.set(cookieName, decrypted)
+    } catch (e) {
+      return new Response(e.message || e.toString(), { status: 500 })
```

Comment:
Same as above, lets throw an error here. Also, this probably shouldn't be a 500 error, as its not a server error. Maybe a `400` of some variety?